app/models/katello/concerns/smart_proxy_extensions.rb

```diff
@@ -71,6 +71,7 @@ def update_puppet_path
         elsif has_feature?(PULP_NODE_FEATURE)
           path = ProxyAPI::PulpNode.new(:url => self.url).capsule_puppet_path['puppet_content_dir']
         end
+        path = '' if path.nil?
```

Comment:
Nitpick, you could do:  

`path ||= ''`

or maybe even be better below that:

`self.update_attribute(:puppet_path, path || '') if persisted?`